### PR TITLE
refactor NewMainKubelet func, extract queryCloudIpsAndNames func

### PR DESCRIPTION
**Pull request by oulinbao**
**What this PR does / why we need it**:
refactor long function and put the new func to cloud.go

**Special notes for your reviewer**:

**Release note**:
```release-note
None
```

### CODE REVIEW

**Review by feiskyer — CHANGES_REQUESTED** [review]
Could you add unit tests for the new function?

**Comment by oulinbao** [comment]
OK，thank you. I will add it tomorrow.


发自网易邮箱大师
On 12/01/2017 22:44, Pengfei Ni wrote:

@feiskyer requested changes on this pull request.

Could you add unit tests for the new function?

**Review by jhorwit2 — CHANGES_REQUESTED** [review]
Thanks for the PR @oulinbao!

I'm not sure this is the best place for this code though. The reason I say that is this is not valuable to the external cloud controller manager since it cannot use the same methods as the kubelet to obtain node information, so this code would be removed once the providers are out-of-tree. The CCM uses methods already on the Instances interface to accomplish this and I don't think we should combine kubelet specific logic into the cloud provider that can't be reused by the CCM. 

cc @wlan0 @luxas

**Comment by oulinbao** [comment]
@feiskyer, I have added unit test for this function.
@jhorwit2, I have move the new function back to kubelet.go.
Thanks a lot to you.

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -328,6 +328,43 @@ func getRuntimeAndImageServices(remoteRuntimeEndpoint string, remoteImageEndpoin
 	return rs, is, err
 }
 
+// queryVariablesFromCloud query nodeName, cloudIPs, cloudNames from cloud, these params are both in and out params
+func queryVariablesFromCloud(cloud cloudprovider.Interface, hostName string, nodeName *types.NodeName, cloudIPs *[]net.IP, cloudNames *[]string) (err error) {
```
Comment (jhorwit2) [inline pkg/kubelet/kubelet.go]: The in/out parameters aren't a common pattern used in Kubernetes. We should use multiple return parameters instead. 

`queryVariablesFromCloud` is a little vague for a function name.

**Review by jhorwit2 — CHANGES_REQUESTED** [review]

**Comment by oulinbao** [comment]
@jhorwit2  I have changed the in/out parameters to return value format.

**Comment by jhorwit2** [comment]
@oulinbao you need to run `hack/update-bazel.sh`

**Comment by oulinbao** [comment]
OK，thanks.




发自网易邮箱大师
On 12/04/2017 13:54, Josh Horwitz wrote:

@oulinbao you need to run hack/update-bazel.sh

**Comment by oulinbao** [comment]
@feiskyer, I have added unit test for this function.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Pull request closed without merging**